src/main/scala/cromwell/engine/io/gcs/GcsFileSystemProvider.scala

```diff
@@ -0,0 +1,145 @@
+package cromwell.engine.io.gcs
+
+import java.io.{FileNotFoundException, OutputStream}
+import java.net.URI
+import java.nio.channels.{Channels, SeekableByteChannel}
+import java.nio.file.DirectoryStream.Filter
+import java.nio.file._
+import java.nio.file.attribute.{BasicFileAttributes, FileAttribute, FileAttributeView}
+import java.nio.file.spi.FileSystemProvider
+import java.util
+import java.util.Collections
+import java.util.concurrent.{AbstractExecutorService, TimeUnit}
+
+import com.google.api.services.storage.model.StorageObject
+import com.google.cloud.hadoop.gcsio.{GoogleCloudStorageReadChannel, GoogleCloudStorageWriteChannel, ObjectWriteConditions}
+import com.google.cloud.hadoop.util.{ApiErrorExtractor, AsyncWriteChannelOptions, ClientRequestHelper}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutorService}
+
+object GcsFileSystemProvider {
+  def getInstance(gcsInterface: GoogleCloudStorage)(implicit executionContext: ExecutionContext) = new GcsFileSystemProvider(gcsInterface, executionContext)
+}
+
+/**
+  * Converts a Scala ExecutionContext to a Java ExecutorService.
```

Comment:
I think this is where I found it:
https://groups.google.com/forum/#!topic/scala-user/ZyHrfzD7eX8